### PR TITLE
Ensure analyzer package is referenced by Microsoft.AspNetCore.Mvc

### LINKED ISSUE #7684: Create an analyzer to warn users not to decorate filters on page handler methods
Related discussion: https://github.com/aspnet/Mvc/issues/7674

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #7684

### CODE REVIEW

**Comment by pranavkm** [comment]
<details>
 <summary>Nuspec</summary>
 <p>

  ```xml
    <?xml version="1.0" encoding="utf-8"?>
    <package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
      <metadata>
        <id>Microsoft.AspNetCore.Mvc</id>
        <version>2.2.0-preview1-t000</version>
        <authors>Microsoft</authors>
        <owners>Microsoft</owners>
        <requireLicenseAcceptance>true</requireLicenseAcceptance>
        <licenseUrl>https://raw.githubusercontent.com/aspnet/Home/2.0.0/LICENSE.txt</licenseUrl>
        <projectUrl>https://asp.net/</projectUrl>
        <iconUrl>https://go.microsoft.com/fwlink/?LinkID=288859</iconUrl>
        <description>ASP.NET Core MVC is a web framework that gives you a powerful, patterns-based way to build dynamic websites and web APIs. ASP.NET Core MVC enables a clean separation of concerns and gives you full control over markup.
    This package was built from the source at:
    https://github.com/aspnet/Mvc/tree/975dff9c3a6c5d31ebc88bbba0406cd3a8d37582</description>
        <copyright>© Microsoft Corporation. All rights reserved.</copyright>
        <tags>aspnetcore aspnetcoremvc</tags>
        <repository type="git" url="https://github.com/aspnet/Mvc" commit="975dff9c3a6c5d31ebc88bbba0406cd3a8d37582" />
        <dependencies>
          <group targetFramework=".NETStandard2.0">
            <dependency id="Microsoft.AspNetCore.Mvc.Analyzers" version="2.2.0-preview1-t000" include="All" />
            <dependency id="Microsoft.AspNetCore.Mvc.ApiExplorer" version="2.2.0-preview1-t000" exclude="Build,Analyzers" />
            <dependency id="Microsoft.AspNetCore.Mvc.Cors" version="2.2.0-preview1-t000" exclude="Build,Analyzers" />
            <dependency id="Microsoft.AspNetCore.Mvc.DataAnnotations" version="2.2.0-preview1-t000" exclude="Build,Analyzers" />
            <dependency id="Microsoft.AspNetCore.Mvc.Formatters.Json" version="2.2.0-preview1-t000" exclude="Build,Analyzers" />
            <dependency id="Microsoft.AspNetCore.Mvc.Localization" version="2.2.0-preview1-t000" exclude="Build,Analyzers" />
            <dependency id="Microsoft.AspNetCore.Mvc.RazorPages" version="2.2.0-preview1-t000" exclude="Build,Analyzers" />
            <dependency id="Microsoft.AspNetCore.Mvc.TagHelpers" version="2.2.0-preview1-t000" exclude="Build,Analyzers" />
            <dependency id="Microsoft.AspNetCore.Mvc.ViewFeatures" version="2.2.0-preview1-t000" exclude="Build,Analyzers" />
            <dependency id="Microsoft.AspNetCore.Mvc.Razor.Extensions" version="2.2.0-preview1-34481" include="All" />
            <dependency id="Microsoft.AspNetCore.Razor.Design" version="2.2.0-preview1-34481" include="All" />
            <dependency id="Microsoft.Extensions.Caching.Memory" version="2.2.0-preview1-34481" exclude="Build,Analyzers" />
            <dependency id="Microsoft.Extensions.DependencyInjection" version="2.2.0-preview1-34481" exclude="Build,Analyzers" />
          </group>
        </dependencies>
      </metadata>
      <files>
        <file src="D:\work\Mvc\src\Microsoft.AspNetCore.Mvc\bin\Release\netstandard2.0\Microsoft.AspNetCore.Mvc.dll" target="lib\netstandard2.0\Microsoft.AspNetCore.Mvc.dll" />
        <file src="D:\work\Mvc\src\Microsoft.AspNetCore.Mvc\bin\Release\netstandard2.0\Microsoft.AspNetCore.Mvc.xml" target="lib\netstandard2.0\Microsoft.AspNetCore.Mvc.xml" />
      </files>
    </package>
  ```
 </p>
</details>

<details>
 <summary>Dll references</summary>
 <p>

  ```
  // Detected Target-Framework-Id: .NETStandard,Version=v2.0

 // Referenced assemblies (in metadata order):
 // netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 (unresolved)
 // Microsoft.AspNetCore.Mvc.Core, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60
 	// Assembly reference loading information:
 	// Info: Success - Found in Assembly List
 
 // Microsoft.Extensions.DependencyInjection.Abstractions, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60 (unresolved)
 // Microsoft.AspNetCore.Mvc.ApiExplorer, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60 (unresolved)
 // Microsoft.AspNetCore.Mvc.ViewFeatures, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60 (unresolved)
 // Microsoft.AspNetCore.Mvc.Razor, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60 (unresolved)
 // Microsoft.AspNetCore.Mvc.RazorPages, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60 (unresolved)
 // Microsoft.AspNetCore.Mvc.TagHelpers, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60 (unresolved)
 // Microsoft.AspNetCore.Mvc.DataAnnotations, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60 (unresolved)
 // Microsoft.AspNetCore.Mvc.Formatters.Json, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60 (unresolved)
 // Microsoft.AspNetCore.Mvc.Cors, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60 (unresolved)
 // Microsoft.Extensions.Options, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60 (unresolved)

  ```
 </p>
</details>

File: src/Microsoft.AspNetCore.Mvc.Analyzers.Experimental/ActionsMustNotBeAsyncVoidAnalyzer.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ public class ActionsMustNotBeAsyncVoidAnalyzer : ControllerAnalyzerBase
         public static readonly string ReturnTypeKey = "ReturnType";
 
         public ActionsMustNotBeAsyncVoidAnalyzer()
-            : base(DiagnosticDescriptors.MVC7003_ActionsMustNotBeAsyncVoid)
+            : base(ExperimentalDiagnosticDescriptors.MVC7003_ActionsMustNotBeAsyncVoid)
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Analyzers.Experimental/ActionsMustNotBeAsyncVoidAnalyzer.cs]: Some diamond dependency shennanigans. Renaming the type in the experimental package

File: test/Microsoft.AspNetCore.Mvc.Test/ApplicationParts/ApplicationAssembliesProviderTest.cs [hunk 2]
```diff
@@ -22,6 +22,7 @@ public void ReferenceAssemblies_ReturnsLoadableReferenceAssemblies()
             // Arrange
             var excludeAssemblies = new string[]
             {
+                "Microsoft.AspNetCore.Mvc.Analyzers",
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Test/ApplicationParts/ApplicationAssembliesProviderTest.cs]: The package reference is in the deps file, so this is expected.

File: src/Microsoft.AspNetCore.Mvc/Microsoft.AspNetCore.Mvc.csproj [hunk 3]
```diff
@@ -9,7 +9,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\Microsoft.AspNetCore.Mvc.Analyzers\Microsoft.AspNetCore.Mvc.Analyzers.csproj" PrivateAssets="None" ReferenceOutputAssembly="false" />
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc/Microsoft.AspNetCore.Mvc.csproj]: Actual change

File: src/Microsoft.AspNetCore.Mvc.Analyzers.Experimental/ActionsMustNotBeAsyncVoidAnalyzer.cs (hunk 1 above)
Comment (ajaybhargavb) [reply src/Microsoft.AspNetCore.Mvc.Analyzers.Experimental/ActionsMustNotBeAsyncVoidAnalyzer.cs]: Where else does `DiagnosticDescriptors` exist? What is it clashing with?

Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.Analyzers.Experimental/ActionsMustNotBeAsyncVoidAnalyzer.cs]: https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Analyzers/DiagnosticDescriptors.cs.

**Pull request merged**